routing/rt/rt.go

```diff
@@ -0,0 +1,324 @@
+		rt.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info.Copy())
+	}
+}
+
+func (rt *RoutingTable) Copy() *RoutingTable {
+	// TODO: add tests
+	channels := rt.AllChannels()
+	newRT := NewRoutingTable()
+	newRT.AddNode(rt.Nodes()...)
+	for _, channel := range channels {
+		newRT.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info.Copy())
+	}
+	return newRT
+}
+
+func (rt *RoutingTable) AddTableFiltered(rt2 *RoutingTable, Node graph.ID, MaxDist int) {
+	// newNodes := rt2.Nodes()
+	rtCopy := rt.Copy()
+	rtCopy.AddTable(rt2)
+	distances, _, err := rtCopy.G.Bfs(Node)
+	if err != nil {
+		panic(err)
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	newChannels := rt2.AllChannels()
+	for _, channel := range newChannels {
+		if !rt.HasChannel(channel.Id1, channel.Id2, channel.EdgeID) && IsClose(channel.Id1, channel.Id2) {
+			rt.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info)
+		}
+	}
+}
+
+func (rt *RoutingTable) ApplyDiffBuffFiltered(diffBuff *DifferenceBuffer, Node graph.ID, MaxDist int) {
+	rtCopy := rt.Copy()
+	rtCopy.ApplyDiffBuff(diffBuff)
+	distances, _, err := rtCopy.G.Bfs(Node)
+	if err != nil {
+		panic(err)
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	filteredDiffBuff := NewDifferenceBuffer()
+	for _, chOp := range *diffBuff {
+		if chOp.Operation == REMOVE_CHANNEL || (chOp.Operation == ADD_CHANNEL && IsClose(chOp.Id1, chOp.Id2)) {
+			*filteredDiffBuff = append(*filteredDiffBuff, chOp)
+		}
+	}
+	rt.ApplyDiffBuff(filteredDiffBuff)
+}
+
+func AddressDistance(x1, x2 graph.ID) int {
+	// TODO: INSERT CORRECT FUNCTION. It is important only for beacon-related functionality.
+	return 1
+}
+
+func (rt *RoutingTable) ClosestAddressNode(dest graph.ID, exclude map[graph.ID]struct{}) *graph.ID {
+	found := false
+	var x graph.ID
+	var nodeId graph.ID
+	g := rt.G
+	for _, nodeId = range g.GetIDs() {
+		if exclude != nil {
+			if _, ok := exclude[nodeId]; ok {
+				continue
+			}
+		}
+		if !found || AddressDistance(dest, x) > AddressDistance(dest, nodeId) {
+			x = nodeId
+			found = true
+		}
+	}
+	if found {
+		return &x
+	} else {
+		return nil
+	}
+}
+
+func (rt *RoutingTable) String() string {
+	rez := ""
+	edges := rt.G.GetUndirectedEdges()
+	for _, edge := range edges {
+		rez += fmt.Sprintf("%v %v %v", edge.Source(), edge.Target(), edge.Info())
+	}
+	return rez
+}
+
+// Find k-shortest path between source and destination
+func (rt *RoutingTable) KSP(src, dst graph.ID, k int) ([][]graph.ID, error) {
+	return graph.KSP(rt.G, src, dst, k)
+}
+
+// Returns copy of channel info for channel in routing table
+func (rt *RoutingTable) GetChannelInfo(id1, id2 graph.ID, edgeId graph.EdgeID) (*graph.ChannelInfo, error) {
```

Comment:
I don't think this function is needed since the graph itself is currently publicly exported.